.github/workflows/lint.yml

```diff
@@ -9,14 +9,10 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - name: Set local Go version
-        run: |
-          VERSION=`cat .go-version| awk '{printf$1}'`
-          echo "go_version=$VERSION" >> $GITHUB_ENV
       - name: Setup Go Environment
         uses: actions/setup-go@v3
         with:
-          go-version: "${{ env.go_version }}"
+          go-version-file: ".go-version"
```

Comment:
While I personally use `goenv` I think we should ultimately rely on the version described in `go.mod` for the workflow. Whenever we upgrade the go version we'll modify `go.mod` rather than calling `goenv local`. Furthermore, it's how we have it for [workflows in go-tfe](https://github.com/hashicorp/go-tfe/blob/736b3590db21ee9cc3f2615365662787ffcbe24a/.github/workflows/ci.yml#L15-L19).

Reply:
```suggestion
          go-version-file: "go.mod"
```